tests/src/test/scala/cats/tests/OptionTTests.scala

```diff
@@ -1,11 +1,110 @@
+    checkAll("Order[OptionT[ListWrapper, Int]]", SerializableTests.serializable(Order[OptionT[ListWrapper, Int]]))
+
+    PartialOrder[OptionT[ListWrapper, Int]]
+    Eq[OptionT[ListWrapper, Int]]
+  }
+
+  {
+    // F has a Functor
+    implicit val F = ListWrapper.functor
+
+    checkAll("OptionT[ListWrapper, Int]", FunctorTests[OptionT[ListWrapper, ?]].functor[Int, Int, Int])
+    checkAll("Functor[OptionT[ListWrapper, ?]]", SerializableTests.serializable(Functor[OptionT[ListWrapper, ?]]))
+  }
+
+  {
+    // F has a Monad
+    implicit val F = ListWrapper.monad
+    implicit val eq0 = OptionT.catsDataEqForOptionT[ListWrapper, Option[Int]]
+    implicit val eq1 = OptionT.catsDataEqForOptionT[OptionT[ListWrapper, ?], Int](eq0)
+
+    checkAll("OptionT[ListWrapper, Int]", MonadTests[OptionT[ListWrapper, ?]].monad[Int, Int, Int])
+    checkAll("Monad[OptionT[ListWrapper, ?]]", SerializableTests.serializable(Monad[OptionT[ListWrapper, ?]]))
+
+    // checkAll("OptionT[ListWrapper, Int]", MonadFilterTests[OptionT[ListWrapper, ?]].monadFilter[Int, Int, Int])
+    // checkAll("MonadFilter[WriterT[ListWrapper, ListWrapper[Int], ?]]", SerializableTests.serializable(MonadFilter[OptionT[ListWrapper, ?]]))
+
+    // checkAll("OptionT[ListWrapper, Int]", MonadCombineTests[OptionT[ListWrapper, ?]].monadCombine[Int, Int, Int])
+    // checkAll("MonadCombine[OptionT[ListWrapper, ?]]", SerializableTests.serializable(MonadCombine[OptionT[ListWrapper, ?]]))
+
+    checkAll("OptionT[ListWrapper, Int]", SemigroupKTests[OptionT[ListWrapper, ?]].semigroupK[Int])
+    checkAll("SemigroupK[OptionT[ListWrapper, ?]]", SerializableTests.serializable(SemigroupK[OptionT[ListWrapper, ?]]))
+
+    checkAll("OptionT[ListWrapper, Int]", MonoidKTests[OptionT[ListWrapper, ?]].monoidK[Int])
+    checkAll("MonoidK[OptionT[ListWrapper, ?]]", SerializableTests.serializable(MonoidK[OptionT[ListWrapper, ?]]))
+
+    FlatMap[OptionT[ListWrapper, ?]]
+    // Alternative[OptionT[ListWrapper, ?]]
+    Applicative[OptionT[ListWrapper, ?]]
+    Apply[OptionT[ListWrapper, ?]]
+    Functor[OptionT[ListWrapper, ?]]
+    MonoidK[OptionT[ListWrapper, ?]]
+    SemigroupK[OptionT[ListWrapper, ?]]
+  }
+
+  {
+    // F has a MonadRec
+    implicit val F = ListWrapper.monadRec
+
+    checkAll("OptionT[ListWrapper, Int]", MonadRecTests[OptionT[ListWrapper, ?]].monadRec[Int, Int, Int])
+    checkAll("MonadRec[OptionT[ListWrapper, ?]]", SerializableTests.serializable(MonadRec[OptionT[ListWrapper, ?]]))
+
+    Monad[OptionT[ListWrapper, ?]]
+    FlatMap[OptionT[ListWrapper, ?]]
+    Applicative[OptionT[ListWrapper, ?]]
+    Apply[OptionT[ListWrapper, ?]]
+    Functor[OptionT[ListWrapper, ?]]
+  }
+
+  {
+    // F has a MonadError
+    type SXor[A] = String Xor A
+
+    implicit val monadError = OptionT.catsDataMonadErrorForOptionT[SXor, String]
+
+    // checkAll("OptionT[String Xor ?, Int]", MonadErrorTests[OptionT[SXor, ?], String].monadError[Int, Int, Int])
```

Comment:
I need an `Arbitrary[λ[(α, β) => OptionT[String Xor α, β]]]` which I don't seem to get from the `Arbitrary` instances of `Xor` and `OptionT`.

I need to figure how to create this instance and if it has a place in `laws.discipline.arbitrary`.